src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,638 @@
+          if ('odd-segments' === trackDisplayMode && 0 !== (index + 1) % 2) {
+            return (
+              <TrackSegment
+                key={index}
+                leftPercent={leftPercent}
+                rightPercent={rightPercent}
+              />
+            );
+          } else if (
+            'even-segments' === trackDisplayMode &&
+            0 === (index + 1) % 2
+          ) {
+            return (
+              <TrackSegment
+                key={index}
+                leftPercent={leftPercent}
+                rightPercent={rightPercent}
+              />
+            );
+          }
+          return null;
+        })}
+    </>
+  );
+};
+
+/** Thumb is a local component used to show and modify the values maintained by the [[Slider]].
+ * Only one Thumb can be active at a time. A Thumb is made active when the user selects
+ * it with pointer. Whenever a Thumb is active, focused, or hovered its tooltip is shown. */
+const Thumb = ({ index, isActive }: { index: number; isActive: boolean }) => {
+  const {
+    hideTooltip,
+    railDomain,
+    setThumbMoveIndex,
+    toolTipFunction,
+    step,
+    values,
+    setThumbValue,
+    thumbMode,
+    toolTipPosition,
+  } = useSliderContext();
+  const thumbRef = React.useRef<HTMLDivElement>(null);
+  const handleOnKeyDown = React.useCallback(
+    (event: React.KeyboardEvent<HTMLDivElement>) => {
+      const [minVal, maxVal] = getAllowableThumbRange(
+        thumbMode,
+        values,
+        index,
+        railDomain,
+        step,
+      );
+
+      if (event.key === 'ArrowLeft') {
+        const newValue = Math.max(values[index] - step, minVal);
+        setThumbValue(index, newValue);
+      } else if (event.key === 'ArrowRight') {
+        const newValue = Math.min(values[index] + step, maxVal);
+        setThumbValue(index, newValue);
+      } else if (event.key === 'Home') {
+        setThumbValue(index, minVal);
+      } else if (event.key === 'End') {
+        setThumbValue(index, maxVal);
+      }
+    },
+    [thumbMode, values, index, railDomain, step, setThumbValue],
+  );
+
+  const handlePointerDownOnThumb = React.useCallback(
+    (event: React.PointerEvent) => {
+      thumbRef.current?.focus();
+      setThumbMoveIndex(index);
+      event.preventDefault();
+      event.stopPropagation();
+    },
+    [index, setThumbMoveIndex],
+  );
+
+  const [hasFocus, setHasFocus] = React.useState(false);
+  const [isHovered, setIsHovered] = React.useState(false);
+
+  const generateTooltip = React.useCallback(
+    (val: number): string => {
+      return toolTipFunction
+        ? toolTipFunction(val, step)
+        : formatNumberValue(val, step);
+    },
+    [toolTipFunction, step],
+  );
+
+  const currentValue = values[index];
+  const tooltipContent = generateTooltip(currentValue);
+  const leftPercent =
+    (100.0 * (currentValue - railDomain.min)) /
+    (railDomain.max - railDomain.min);
+
+  const thumbClassNames = React.useMemo(
```

Comment:
I think it just adds additional overhead with the useMemo than actual value.. Have a simple inline cx with the object.. It is on a div element, so it is not something heavy you need to handle here...

Reply:
useMemo is no longer used here.